node_helper.js

```diff
@@ -21,7 +21,11 @@ module.exports = NodeHelper.create({
       this.relay.writeSync(this.config.relayOnState);
     }
     else if (this.config.relayPIN == false){
-      exec("/opt/vc/bin/tvservice --preferred && sudo chvt 6 && sudo chvt 7", null);
+      // Check if hdmi output is already on
+      exec("/opt/vc/bin/tvservice -s").stdout.on('data', function(data) {
+        if (data.indexOf("0x120002") !== -1)
```

Comment:
With `tvservice -s` you get the status of video output. `0x12002` indicates that the output is already on. Turning on, when hdmi output is already on, leads to flickering.